src/NLog/Targets/ConsoleWordHighlightingRule.cs

```diff
@@ -116,7 +130,7 @@ public Regex CompiledRegex
                         }
                     }
 
-                    RegexOptions regexOptions = RegexOptions.Compiled;
+                    RegexOptions regexOptions = RegexOptions.None;
```

Comment:
I think we should pick one of the two: use caching, or use compied. As this one isn't 1 of them, I prefer a revert (also because of the name).